Dockerfile

```diff
@@ -0,0 +1,20 @@
+# Build Stage
+FROM --platform=linux/amd64 ubuntu:20.04 as builder
+
+## Install build dependencies.
+RUN apt-get update && \
+    DEBIAN_FRONTEND=noninteractive apt-get install -y cmake clang
+
+## Add source code to the build stage.
+ADD . /jpeg-encoder
+WORKDIR /jpeg-encoder
+
+## TODO: ADD YOUR BUILD INSTRUCTIONS HERE.
+RUN mkdir build && cd build && clang++ ../fuzz-jpeg.cpp ../jpeg_encoder.cpp -o fuzzjpeg -lstdc++

```

Comment:
If you are using clang already, is there a reason you don't create a libFuzzer target?

Reply:
I updated this to have a much smaller file size and changed the folder to testsuite. With regards to why I didn't use libfuzzer, I'm honestly not as familiar with libfuzzer and in order to use both the readFromBMP (which takes a file input) and encodeToJPG functions in one go for more potential coverage, I wanted to use the file input with @@ from mayhem instead.

Reply:
Ok. I may add a sanitizer here to see if we can get better results, although your most recent run looks a lot better than the first one:
https://mayhem.forallsecure.com/jbgreenh/jpeg-encoder/fuzzjpeg/2